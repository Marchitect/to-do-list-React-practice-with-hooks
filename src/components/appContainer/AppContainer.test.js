// AppContainer.test.js
import React from "react"
import { AppContainer } from "./AppContainer.component"
import { render, fireEvent } from "@testing-library/react"

test("AppContainer renders correctly", () => {
  const appContainer = render(<AppContainer />)
  expect(appContainer).toMatchSnapshot()
})

test("When the Submit Button is clicked, the new to-do item is added", () => {
  const { getByText, getByPlaceholderText } = render(
    <AppContainer />
  )
  expect(getByText("Submit")).toBeTruthy()
  const input = getByPlaceholderText("to-do")
  fireEvent.change(input, { target: { value: "test item" } })
  fireEvent.click(getByText("Submit"))
})

test("When the Complete Button is clicked, the select to-do item is removed", () => {
  const { getAllByText } = render(<AppContainer />)
  expect(getAllByText("Complete")[0]).toBeTruthy()
  fireEvent.click(getAllByText("Complete")[0])
})
