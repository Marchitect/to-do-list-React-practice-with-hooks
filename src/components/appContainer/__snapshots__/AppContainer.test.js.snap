// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxmLO exQSXV"
      >
        <div>
          <form
            class="sc-AxhCb gqRNTI"
          >
            <label
              class="sc-AxhUy fPIdVO"
              for="todo-item-input"
            >
              Enter your to-do item here:
            </label>
            <div
              class="sc-AxjAm ejbBdx"
            >
              <input
                class="sc-AxirZ TIGOt"
                id="todo_input"
                name="todo-item-input"
                placeholder="to-do"
                type="text"
                value=""
              />
              <button
                class="sc-AxiKw iWkUVY"
                id="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
        <div
          class="sc-Axmtr gbpYMa"
        >
          <div
            class="sc-AxgMl edRssr"
          >
            Learn React state hook
          </div>
          <button
            class="sc-AxheI kluNOV"
          >
            Complete
          </button>
        </div>
        <div
          class="sc-Axmtr gbpYMa"
        >
          <div
            class="sc-AxgMl edRssr"
          >
            Meet friends for lunch
          </div>
          <button
            class="sc-AxheI kluNOV"
          >
            Complete
          </button>
        </div>
        <div
          class="sc-Axmtr gbpYMa"
        >
          <div
            class="sc-AxgMl edRssr"
          >
            Learn React effect hook
          </div>
          <button
            class="sc-AxheI kluNOV"
          >
            Complete
          </button>
        </div>
        <div
          class="sc-Axmtr gbpYMa"
        >
          <div
            class="sc-AxgMl edRssr"
          >
            Buy groceries
          </div>
          <button
            class="sc-AxheI kluNOV"
          >
            Complete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxmLO exQSXV"
    >
      <div>
        <form
          class="sc-AxhCb gqRNTI"
        >
          <label
            class="sc-AxhUy fPIdVO"
            for="todo-item-input"
          >
            Enter your to-do item here:
          </label>
          <div
            class="sc-AxjAm ejbBdx"
          >
            <input
              class="sc-AxirZ TIGOt"
              id="todo_input"
              name="todo-item-input"
              placeholder="to-do"
              type="text"
              value=""
            />
            <button
              class="sc-AxiKw iWkUVY"
              id="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-Axmtr gbpYMa"
      >
        <div
          class="sc-AxgMl edRssr"
        >
          Learn React state hook
        </div>
        <button
          class="sc-AxheI kluNOV"
        >
          Complete
        </button>
      </div>
      <div
        class="sc-Axmtr gbpYMa"
      >
        <div
          class="sc-AxgMl edRssr"
        >
          Meet friends for lunch
        </div>
        <button
          class="sc-AxheI kluNOV"
        >
          Complete
        </button>
      </div>
      <div
        class="sc-Axmtr gbpYMa"
      >
        <div
          class="sc-AxgMl edRssr"
        >
          Learn React effect hook
        </div>
        <button
          class="sc-AxheI kluNOV"
        >
          Complete
        </button>
      </div>
      <div
        class="sc-Axmtr gbpYMa"
      >
        <div
          class="sc-AxgMl edRssr"
        >
          Buy groceries
        </div>
        <button
          class="sc-AxheI kluNOV"
        >
          Complete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
